Lib/multiprocessing/pool.py

```diff
@@ -294,8 +296,15 @@ def _join_exited_workers(pool):
             if worker.exitcode is not None:
                 # worker exited
                 util.debug('cleaning up worker %d' % i)
+                pid =  worker.ident
                 worker.join()
                 cleaned = True
+                if pid in job_assignments:
+                    # If the worker process died without communicating back
+                    # while running a job, add a default result for it.
+                    job = job_assignments[pid]
+                    outqueue.put((job, i, (False, RuntimeError("Worker died"))))
```

Comment:
Can we put the exit code into the error? And presumably this should at least be a `ProcessError` if not a new subclass (`WorkerExitedError`?).

Reply:
Documenting the exception is probably sufficient documentation as well.